src/wp-includes/functions.php

```diff
@@ -8221,21 +8228,49 @@ function recurse_dirsize( $directory, $exclude = null, $max_execution_time = nul
  * Removes the current directory and all parent directories from the `dirsize_cache` transient.
  *
  * @since 5.6.0
+ * @since 5.9.0 Added input validation with a doing it wrong for invalid input.
  *
  * @param string $path Full path of a directory or file.
  */
 function clean_dirsize_cache( $path ) {
+	if ( ! is_string( $path ) || empty( $path ) ) {
+		_doing_it_wrong(
```

Comment:
WordPress generally tends to avoid function names in translatable strings, as they don't need translation and would be confusing if accidentally translated.

I would suggest something like this here for consistency with other similar messages:
```
		_doing_it_wrong(
			__FUNCTION__,
			sprintf(
				/* translators: 1: Function name, 2: A variable type, like "boolean" or "integer". */
				__( '%1$s only accepts a non-empty path string, received %2$s.' ),
				'<code>clean_dirsize_cache()</code>',
				'<code>' . gettype( $path ) . '</code>'
			),
			'5.9.0'
		);
```